diff -ru libtheora-1.1.1.org/configure libtheora-1.1.1/configure
--- libtheora-1.1.1.org/configure	2009-10-01 20:04:08.000000000 +0200
+++ libtheora-1.1.1/configure	2014-04-05 00:00:32.000000000 +0200
@@ -11863,7 +11863,7 @@
         case $host in
         *)
                 DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
-                CFLAGS="-Wall -Wno-parentheses -O3 -fforce-addr -fomit-frame-pointer -finline-functions -funroll-loops"
+                CFLAGS="-Wall -Wno-parentheses -O3 -fomit-frame-pointer -finline-functions -funroll-loops"
                 PROFILE="-Wall -Wno-parentheses -pg -g -O3 -fno-inline-functions -DDEBUG";;
         esac
 fi
