diff -ru glew-1.7.0.org/include/GL/glew.h glew-1.7.0/include/GL/glew.h
--- glew-1.7.0.org/include/GL/glew.h	2011-08-25 16:17:55.000000000 +0200
+++ glew-1.7.0/include/GL/glew.h	2012-01-29 15:37:10.000000000 +0100
@@ -109,7 +109,7 @@
 #define GLEW_APIENTRY_DEFINED
 #  if defined(__MINGW32__) || defined(__CYGWIN__)
 #    define APIENTRY __stdcall
-#  elif (_MSC_VER >= 800) || defined(_STDCALL_SUPPORTED) || defined(__BORLANDC__)
+#  elif (defined(_MSC_VER) && (_MSC_VER >= 800)) || defined(_STDCALL_SUPPORTED) || defined(__BORLANDC__)
 #    define APIENTRY __stdcall
 #  else
 #    define APIENTRY
diff -ru glew-1.7.0.org/Makefile glew-1.7.0/Makefile
--- glew-1.7.0.org/Makefile	2011-08-25 16:17:55.000000000 +0200
+++ glew-1.7.0/Makefile	2012-01-29 15:36:23.000000000 +0100
@@ -54,9 +54,9 @@
 DIST_SRC_ZIP = glew-$(GLEW_VERSION).zip
 DIST_SRC_TGZ = glew-$(GLEW_VERSION).tgz
 
-AR = ar
+AR?=ar
 INSTALL = install
-STRIP = strip
+STRIP?=strip
 RM = rm -f
 LN = ln -sf
 ifeq ($(MAKECMDGOALS), debug)
@@ -65,7 +65,9 @@
 OPT = $(POPT)
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+# openMSX dedicated build: use flavour's optimization flags instead:
+#CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS += $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
 
 LIB.SRCS = src/glew.c
 LIB.OBJS = $(LIB.SRCS:.c=.o)
@@ -221,7 +223,7 @@
 
 install.bin: glew.bin
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
+#	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
 install.include:
 	$(INSTALL) -d -m 0755 $(INCDIR)
