--- glew-1.4.0.org/Makefile	2007-03-02 17:56:00.000000000 +0100
+++ glew-1.4.0/Makefile	2007-06-17 02:59:31.000000000 +0200
@@ -60,7 +60,9 @@
 STRIP = -s
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+# openMSX dedicated build: use flavour's optimization flags instead:
+#CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS += $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
 
 LIB.SRCS = src/glew.c
 LIB.OBJS = $(LIB.SRCS:.c=.o)
