diff -ru glew-1.5.8.org/include/GL/glew.h glew-1.5.8/include/GL/glew.h
--- glew-1.5.8.org/include/GL/glew.h	2011-01-31 15:17:27.000000000 +0100
+++ glew-1.5.8/include/GL/glew.h	2011-03-25 19:43:53.000000000 +0100
@@ -109,7 +109,7 @@
 #define GLEW_APIENTRY_DEFINED
 #  if defined(__MINGW32__) || defined(__CYGWIN__)
 #    define APIENTRY __stdcall
-#  elif (_MSC_VER >= 800) || defined(_STDCALL_SUPPORTED) || defined(__BORLANDC__)
+#  elif (defined(_MSC_VER) && (_MSC_VER >= 800)) || defined(_STDCALL_SUPPORTED) || defined(__BORLANDC__)
 #    define APIENTRY __stdcall
 #  else
 #    define APIENTRY
diff -ru glew-1.5.8.org/Makefile glew-1.5.8/Makefile
--- glew-1.5.8.org/Makefile	2011-01-31 15:17:27.000000000 +0100
+++ glew-1.5.8/Makefile	2011-03-25 21:53:14.000000000 +0100
@@ -53,9 +53,9 @@
 DIST_SRC_ZIP = glew-$(GLEW_VERSION).zip
 DIST_SRC_TGZ = glew-$(GLEW_VERSION).tgz
 
-AR = ar
+AR?=ar
 INSTALL = install
-STRIP = strip
+STRIP?=strip
 RM = rm -f
 LN = ln -sf
 ifeq ($(MAKECMDGOALS), debug)
@@ -64,7 +64,9 @@
 OPT = $(POPT)
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+# openMSX dedicated build: use flavour's optimization flags instead:
+#CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS += $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
 
 LIB.SRCS = src/glew.c
 LIB.OBJS = $(LIB.SRCS:.c=.o)
@@ -143,15 +145,15 @@
 	$(INSTALL) -m 0644 include/GL/glew.h $(INCDIR)/
 	$(INSTALL) -m 0644 include/GL/glxew.h $(INCDIR)/
 	$(INSTALL) -m 0644 glew.pc $(LIBDIR)/pkgconfig/
+	$(STRIP) -x lib/$(LIB.STATIC)
+	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 	$(INSTALL) -m 0644 lib/$(LIB.DEVLNK) $(LIBDIR)/
 else
-	$(STRIP) -x lib/$(LIB.STATIC)
-	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
 endif
 # utilities
-	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
+#	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
 uninstall:
 	$(RM) $(INCDIR)/wglew.h
