diff -ru glew-2.1.0.org/config/Makefile.linux-mingw-w64 glew-2.1.0/config/Makefile.linux-mingw-w64
--- glew-2.1.0.org/config/Makefile.linux-mingw-w64	2017-07-31 13:25:39.000000000 +0200
+++ glew-2.1.0/config/Makefile.linux-mingw-w64	2019-01-14 23:40:05.627246200 +0100
@@ -11,7 +11,6 @@
 STRIP :=
 LDFLAGS.GL = -lopengl32 -lgdi32 -luser32 -lkernel32
 CFLAGS.EXTRA  += -fno-builtin -fno-stack-protector
-LDFLAGS.EXTRA += -nostdlib
 WARN = -Wall -W
 POPT = -O2
 BIN.SUFFIX = .exe
@@ -19,4 +18,4 @@
 LIB.DEVLNK    = lib$(NAME).dll.a    # for mingw this is the dll import lib
 LIB.SHARED    = $(NAME).dll
 LIB.STATIC    = lib$(NAME).a        # the static lib will be broken
-LDFLAGS.SO    = -shared -soname $(LIB.SONAME) --out-implib lib/$(LIB.DEVLNK)
+LDFLAGS.SO    = -shared -nostdlib -Wl,-soname,$(LIB.SONAME) -Wl,--out-implib,lib/$(LIB.DEVLNK)
