diff -ru glew-1.9.0.orig/include/GL/glew.h glew-1.9.0/include/GL/glew.h
--- glew-1.9.0.orig/include/GL/glew.h	2012-08-06 10:59:08.000000000 +0200
+++ glew-1.9.0/include/GL/glew.h	2013-05-18 12:29:18.000000000 +0200
@@ -113,7 +113,7 @@
 #define GLEW_APIENTRY_DEFINED
 #  if defined(__MINGW32__) || defined(__CYGWIN__)
 #    define APIENTRY __stdcall
-#  elif (_MSC_VER >= 800) || defined(_STDCALL_SUPPORTED) || defined(__BORLANDC__)
+#  elif (defined(_MSC_VER) && (_MSC_VER >= 800)) || defined(_STDCALL_SUPPORTED) || defined(__BORLANDC__)
 #    define APIENTRY __stdcall
 #  else
 #    define APIENTRY
diff -ru glew-1.9.0.orig/Makefile glew-1.9.0/Makefile
--- glew-1.9.0.orig/Makefile	2012-08-06 10:59:08.000000000 +0200
+++ glew-1.9.0/Makefile	2013-05-18 12:28:30.000000000 +0200
@@ -72,7 +72,9 @@
 OPT = $(POPT)
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+# openMSX dedicated build: use flavour's optimization flags instead:
+#CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS += $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
 
 all debug: glew.lib glew.lib.mx glew.bin
 
@@ -262,7 +264,7 @@
 
 install.bin: glew.bin
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
+#	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
 install.include:
 	$(INSTALL) -d -m 0755 $(INCDIR)
