diff -ur libepoxy-1.5.10.orig/src/gen_dispatch.py libepoxy-1.5.10/src/gen_dispatch.py
--- libepoxy-1.5.10.orig/src/gen_dispatch.py	2022-02-17 13:56:12.000000000 +0100
+++ libepoxy-1.5.10/src/gen_dispatch.py	2025-04-27 23:49:50.511181247 +0200
@@ -123,20 +123,16 @@
         #
         # We retain those aliases.  In the x86_64 ABI, the first 6
         # args are stored in 64-bit registers, so the calls end up
-        # being the same despite the different types.  We just need to
-        # add a cast to uintptr_t to shut up the compiler.
+        # being the same despite the different types.
         if arg_type == 'GLhandleARB':
             assert len(self.args) < 6
-            arg_list_name = '(uintptr_t)' + arg_name
-        else:
-            arg_list_name = arg_name
 
         self.args.append((arg_type, arg_name))
         if self.args_decl == 'void':
-            self.args_list = arg_list_name
+            self.args_list = arg_name
             self.args_decl = arg_type + ' ' + arg_name
         else:
-            self.args_list += ', ' + arg_list_name
+            self.args_list += ', ' + arg_name
             self.args_decl += ', ' + arg_type + ' ' + arg_name
 
     def add_provider(self, condition, loader, condition_name):
diff -ur libepoxy-1.5.10.orig/src/dispatch_wgl.c libepoxy-1.5.10/src/dispatch_wgl.c
--- libepoxy-1.5.10.orig/src/dispatch_wgl.c
+++ libepoxy-1.5.10/src/dispatch_wgl.c
@@ -27,6 +27,7 @@
 
 #include "dispatch_common.h"
 
+static INIT_ONCE initialized = INIT_ONCE_STATIC_INIT;
 static bool first_context_current = false;
 static bool already_switched_to_dispatch_table = false;
 
@@ -62,6 +63,29 @@ epoxy_has_wgl_extension(HDC hdc, const char *ext)
     return epoxy_extension_in_string(getext(hdc), ext);
 }
 
+BOOL
+epoxy_initialize_tls(PINIT_ONCE once, PVOID param, PVOID *context)
+{
+    void *data;
+
+    gl_tls_index = TlsAlloc();
+    if (gl_tls_index == TLS_OUT_OF_INDEXES)
+        return FALSE;
+    wgl_tls_index = TlsAlloc();
+    if (wgl_tls_index == TLS_OUT_OF_INDEXES)
+        return FALSE;
+
+    first_context_current = false;
+
+    data = LocalAlloc(LPTR, gl_tls_size);
+    TlsSetValue(gl_tls_index, data);
+
+    data = LocalAlloc(LPTR, wgl_tls_size);
+    TlsSetValue(wgl_tls_index, data);
+
+    return TRUE;
+}
+
 /**
  * Does the work necessary to update the win32 per-thread dispatch
  * tables when wglMakeCurrent() is called.
@@ -78,6 +102,8 @@ epoxy_handle_external_wglMakeCurrent(void)
     if (!first_context_current) {
         first_context_current = true;
     } else {
+        InitOnceExecuteOnce(&initialized, epoxy_initialize_tls, NULL, NULL);
+
         if (!already_switched_to_dispatch_table) {
             already_switched_to_dispatch_table = true;
             gl_switch_to_dispatch_table();
@@ -103,16 +129,8 @@ DllMain(HINSTANCE dll, DWORD reason, LPVOID reserved)
 
     switch (reason) {
     case DLL_PROCESS_ATTACH:
-        gl_tls_index = TlsAlloc();
-        if (gl_tls_index == TLS_OUT_OF_INDEXES)
-            return FALSE;
-        wgl_tls_index = TlsAlloc();
-        if (wgl_tls_index == TLS_OUT_OF_INDEXES)
-            return FALSE;
-
-        first_context_current = false;
-
-        /* FALLTHROUGH */
+        InitOnceExecuteOnce(&initialized, epoxy_initialize_tls, NULL, NULL);
+        break;
 
     case DLL_THREAD_ATTACH:
         data = LocalAlloc(LPTR, gl_tls_size);
