diff -ur pkg-config-0.29.2.org/glib/glib/goption.c pkg-config-0.29.2/glib/glib/goption.c
--- pkg-config-0.29.2.org/glib/glib/goption.c	2016-03-09 23:25:01
+++ pkg-config-0.29.2/glib/glib/goption.c	2026-09-22 22:38:00
@@ -166,7 +166,7 @@
   gpointer arg_data;
   union
   {
-    gboolean bool;
+    gboolean my_bool;
     gint integer;
     gchar *str;
     gchar **array;
@@ -1600,7 +1600,7 @@
           switch (change->arg_type)
             {
             case G_OPTION_ARG_NONE:
-              *(gboolean *)change->arg_data = change->prev.bool;
+              *(gboolean *)change->arg_data = change->prev.my_bool;
               break;
             case G_OPTION_ARG_INT:
               *(gint *)change->arg_data = change->prev.integer;
