diff -ru portaudio-v190600_20161030.org/Makefile.in portaudio-v190600_20161030/Makefile.in
--- portaudio-v190600_20161030.org/Makefile.in	2016-10-30 02:23:04.000000000 +0100
+++ portaudio-v190600_20161030/Makefile.in	2019-09-23 07:05:01.032185547 +0200
@@ -56,8 +56,7 @@
 	src/common/pa_front.o \
 	src/common/pa_process.o \
 	src/common/pa_stream.o \
-	src/common/pa_trace.o \
-	src/hostapi/skeleton/pa_hostapi_skeleton.o
+	src/common/pa_trace.o
 
 LOOPBACK_OBJS = \
 	qa/loopback/src/audio_analyzer.o \
diff -ru portaudio-v190600_20161030.org/configure portaudio-v190600_20161030/configure
--- portaudio-v190600_20161030.org/configure	2016-10-30 02:23:04.000000000 +0100
+++ portaudio-v190600_20161030/configure	2019-09-23 07:05:46.104421288 +0200
@@ -15276,7 +15276,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lasound  $LIBS"
+LIBS="`$PKG_CONFIG alsa --libs --static`  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -15957,7 +15957,7 @@
             add_objects src/hostapi/wasapi/pa_win_wasapi.o src/common/pa_ringbuffer.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o src/os/win/pa_win_coinitialize.o src/os/win/pa_win_waveformat.o
             LIBS="${LIBS} -lwinmm -lm -lole32 -luuid"
             DLL_LIBS="${DLL_LIBS} -lwinmm -lole32"
-            CFLAGS="$CFLAGS -I\$(top_srcdir)/src/hostapi/wasapi/mingw-include -UPA_USE_WASAPI -DPA_USE_WASAPI=1"
+            CFLAGS="$CFLAGS -UPA_USE_WASAPI -DPA_USE_WASAPI=1"
         fi
         ;;
 
diff -ru portaudio-v190600_20161030.org/src/hostapi/alsa/pa_linux_alsa.c portaudio-v190600_20161030/src/hostapi/alsa/pa_linux_alsa.c
--- portaudio-v190600_20161030.org/src/hostapi/alsa/pa_linux_alsa.c	2016-10-30 02:23:04.000000000 +0100
+++ portaudio-v190600_20161030/src/hostapi/alsa/pa_linux_alsa.c	2019-09-23 19:22:02.033192258 +0200
@@ -762,6 +762,7 @@
         Terminate function.
     */
     /*ENSURE_( snd_lib_error_set_handler(AlsaErrorHandler), paUnanticipatedHostError );*/
+	snd_lib_error_set_handler(NULL);
 
     PA_ENSURE( BuildDeviceList( alsaHostApi ) );
 
diff -ru portaudio-v190600_20161030.org/src/hostapi/wasapi/pa_win_wasapi.c portaudio-v190600_20161030/src/hostapi/wasapi/pa_win_wasapi.c
--- portaudio-v190600_20161030.org/src/hostapi/wasapi/pa_win_wasapi.c	2016-10-30 02:23:04.000000000 +0100
+++ portaudio-v190600_20161030/src/hostapi/wasapi/pa_win_wasapi.c	2019-09-23 07:05:01.040185589 +0200
@@ -55,14 +55,15 @@
 
 // WASAPI
 #include <mmreg.h>  // must be before other Wasapi headers
-#if defined(_MSC_VER) && (_MSC_VER >= 1400)
-	#include <Avrt.h>
+#if defined(_MSC_VER) && (_MSC_VER >= 1400) || defined(__MINGW64_VERSION_MAJOR)
+	#include <avrt.h>
 	#define COBJMACROS
-	#include <Audioclient.h>
+	#include <audioclient.h>
 	#include <endpointvolume.h>
 	#define INITGUID // Avoid additional linkage of static libs, excessive code will be optimized out by the compiler
-	#include <mmdeviceapi.h>
+	#include <functiondiscoverykeys_devpkey.h>
 	#include <functiondiscoverykeys.h>
+	#include <mmdeviceapi.h>
     #include <devicetopology.h>	// Used to get IKsJackDescription interface
 	#undef INITGUID
 #endif
