--- SDL-1.2.12/src/video/quartz/SDL_QuartzEvents.m	2007/07/15 17:22:39	3295
+++ SDL-1.2.12/src/video/quartz/SDL_QuartzEvents.m	2007/08/21 06:52:33	3445
@@ -270,6 +270,8 @@
         [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
         chars = [ event characters ];
         numChars = [ chars length ];
+        if (numChars > 0)
+            [field_edit setString:@""];
     } else {
         numChars = 0;
     }
--- SDL-1.2.12/src/cdrom/macosx/AudioFilePlayer.h.orig	2007-07-05 12:27:56.000000000 -0700
+++ SDL-1.2.12/src/cdrom/macosx/AudioFilePlayer.h	2007-07-05 12:29:36.000000000 -0700
@@ -33,6 +33,7 @@
 #include <CoreServices/CoreServices.h>
 
 #include <AudioUnit/AudioUnit.h>
+#include <AudioUnit/AUNTComponent.h>
 
 #include "SDL_error.h"
 
--- SDL-1.2.12/src/audio/macosx/SDL_coreaudio.c.orig	2007-07-05 12:34:39.000000000 -0700
+++ SDL-1.2.12/src/audio/macosx/SDL_coreaudio.c	2007-07-05 12:35:25.000000000 -0700
@@ -22,6 +22,7 @@
 #include "SDL_config.h"
 
 #include <AudioUnit/AudioUnit.h>
+#include <AudioUnit/AUNTComponent.h>
 
 #include "SDL_audio.h"
 #include "../SDL_audio_c.h"
