Only in SDL-1.2.12.org/include: SDL_config.h
diff -ru SDL-1.2.12.org/src/audio/macosx/SDL_coreaudio.c SDL-1.2.12/src/audio/macosx/SDL_coreaudio.c
--- SDL-1.2.12.org/src/audio/macosx/SDL_coreaudio.c	2007-07-20 07:52:10.000000000 +0200
+++ SDL-1.2.12/src/audio/macosx/SDL_coreaudio.c	2007-12-08 20:57:26.000000000 +0100
@@ -22,6 +22,7 @@
 #include "SDL_config.h"
 
 #include <AudioUnit/AudioUnit.h>
+#include <AudioUnit/AUNTComponent.h>
 
 #include "SDL_audio.h"
 #include "../SDL_audio_c.h"
diff -ru SDL-1.2.12.org/src/cdrom/macosx/AudioFilePlayer.h SDL-1.2.12/src/cdrom/macosx/AudioFilePlayer.h
--- SDL-1.2.12.org/src/cdrom/macosx/AudioFilePlayer.h	2007-07-20 07:52:11.000000000 +0200
+++ SDL-1.2.12/src/cdrom/macosx/AudioFilePlayer.h	2007-12-08 20:57:26.000000000 +0100
@@ -33,6 +33,7 @@
 #include <CoreServices/CoreServices.h>
 
 #include <AudioUnit/AudioUnit.h>
+#include <AudioUnit/AUNTComponent.h>
 
 #include "SDL_error.h"
 
diff -ru SDL-1.2.12.org/src/video/quartz/SDL_QuartzEvents.m SDL-1.2.12/src/video/quartz/SDL_QuartzEvents.m
--- SDL-1.2.12.org/src/video/quartz/SDL_QuartzEvents.m	2007-07-20 07:52:17.000000000 +0200
+++ SDL-1.2.12/src/video/quartz/SDL_QuartzEvents.m	2007-12-08 21:00:16.000000000 +0100
@@ -267,7 +267,9 @@
         the scancode/keysym.
     */
     if (SDL_TranslateUNICODE && state == SDL_PRESSED) {
-        [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
+        if (!([event modifierFlags] & NSCommandKeyMask)) {
+            [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
+        }
         chars = [ event characters ];
         numChars = [ chars length ];
     } else {
@@ -303,7 +305,7 @@
         }
     }
     
-    if (SDL_getenv ("SDL_ENABLEAPPEVENTS"))
+    if (SDL_getenv ("SDL_ENABLEAPPEVENTS") && !(mode_flags & SDL_FULLSCREEN))
         [ NSApp sendEvent:event ];
 }
 
diff -ru SDL-1.2.12.org/src/video/quartz/SDL_QuartzWindow.h SDL-1.2.12/src/video/quartz/SDL_QuartzWindow.h
--- SDL-1.2.12.org/src/video/quartz/SDL_QuartzWindow.h	2007-07-20 07:52:17.000000000 +0200
+++ SDL-1.2.12/src/video/quartz/SDL_QuartzWindow.h	2007-12-08 20:57:26.000000000 +0100
@@ -30,6 +30,7 @@
 - (void)appWillUnhide:(NSNotification*)note;
 - (void)appDidUnhide:(NSNotification*)note;
 - (id)initWithContentRect:(NSRect)contentRect styleMask:(unsigned int)styleMask backing:(NSBackingStoreType)backingType defer:(BOOL)flag;
+- (BOOL)performKeyEquivalent:(NSEvent*)event;
 @end
 
 /* Delegate for our NSWindow to send SDLQuit() on close */
diff -ru SDL-1.2.12.org/src/video/quartz/SDL_QuartzWindow.m SDL-1.2.12/src/video/quartz/SDL_QuartzWindow.m
--- SDL-1.2.12.org/src/video/quartz/SDL_QuartzWindow.m	2007-07-20 07:52:17.000000000 +0200
+++ SDL-1.2.12/src/video/quartz/SDL_QuartzWindow.m	2007-12-08 20:57:26.000000000 +0100
@@ -198,6 +198,14 @@
     return [ super initWithContentRect:contentRect styleMask:styleMask backing:backingType defer:flag ];
 }
 
+- (BOOL)performKeyEquivalent:(NSEvent*)event
+{
+	/* give the menu a chance to handle the key equivalent */
+	[[NSApp mainMenu] performKeyEquivalent:event];
+	/* avoid beep by pretending we handled it */
+	return YES;	
+}
+
 @end
 
 @implementation SDL_QuartzWindowDelegate
