diff -ru SDL-1.2.13.org/configure SDL-1.2.13/configure
--- SDL-1.2.13.org/configure	2007-12-31 06:09:39.000000000 +0100
+++ SDL-1.2.13/configure	2009-05-03 11:28:01.000000000 +0200
@@ -34221,7 +34221,7 @@
     SDL_STATIC_LIBS="\${libdir}/libSDLmain.a \${libdir}/libSDL.a $EXTRA_LDFLAGS"
     ;;
   *)
-    SDL_STATIC_LIBS="$SDL_LIBS"
+    SDL_STATIC_LIBS="$SDL_LIBS $EXTRA_LDFLAGS"
     ;;
 esac
 
diff -ru SDL-1.2.13.org/configure.in SDL-1.2.13/configure.in
--- SDL-1.2.13.org/configure.in	2007-12-31 05:48:39.000000000 +0100
+++ SDL-1.2.13/configure.in	2009-05-03 11:28:14.000000000 +0200
@@ -2749,7 +2749,7 @@
     SDL_STATIC_LIBS="\${libdir}/libSDLmain.a \${libdir}/libSDL.a $EXTRA_LDFLAGS"
     ;;
   *)
-    SDL_STATIC_LIBS="$SDL_LIBS"
+    SDL_STATIC_LIBS="$SDL_LIBS $EXTRA_LDFLAGS"
     ;;
 esac
 
diff -ru SDL-1.2.13.org/src/video/quartz/SDL_QuartzEvents.m SDL-1.2.13/src/video/quartz/SDL_QuartzEvents.m
--- SDL-1.2.13.org/src/video/quartz/SDL_QuartzEvents.m	2007-12-31 07:16:32.000000000 +0100
+++ SDL-1.2.13/src/video/quartz/SDL_QuartzEvents.m	2009-04-29 02:13:53.000000000 +0200
@@ -267,7 +267,9 @@
         the scancode/keysym.
     */
     if (SDL_TranslateUNICODE && state == SDL_PRESSED) {
-        [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
+        if (!([event modifierFlags] & NSCommandKeyMask)) {
+            [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
+        }
         chars = [ event characters ];
         numChars = [ chars length ];
         if (numChars > 0)
@@ -305,7 +307,7 @@
         }
     }
     
-    if (SDL_getenv ("SDL_ENABLEAPPEVENTS"))
+    if (SDL_getenv ("SDL_ENABLEAPPEVENTS") && !(mode_flags & SDL_FULLSCREEN))
         [ NSApp sendEvent:event ];
 }
 
diff -ru SDL-1.2.13.org/src/video/quartz/SDL_QuartzWindow.h SDL-1.2.13/src/video/quartz/SDL_QuartzWindow.h
--- SDL-1.2.13.org/src/video/quartz/SDL_QuartzWindow.h	2007-12-31 05:48:04.000000000 +0100
+++ SDL-1.2.13/src/video/quartz/SDL_QuartzWindow.h	2009-04-29 02:13:53.000000000 +0200
@@ -30,6 +30,7 @@
 - (void)appWillUnhide:(NSNotification*)note;
 - (void)appDidUnhide:(NSNotification*)note;
 - (id)initWithContentRect:(NSRect)contentRect styleMask:(unsigned int)styleMask backing:(NSBackingStoreType)backingType defer:(BOOL)flag;
+- (BOOL)performKeyEquivalent:(NSEvent*)event;
 @end
 
 /* Delegate for our NSWindow to send SDLQuit() on close */
diff -ru SDL-1.2.13.org/src/video/quartz/SDL_QuartzWindow.m SDL-1.2.13/src/video/quartz/SDL_QuartzWindow.m
--- SDL-1.2.13.org/src/video/quartz/SDL_QuartzWindow.m	2007-12-31 05:48:04.000000000 +0100
+++ SDL-1.2.13/src/video/quartz/SDL_QuartzWindow.m	2009-04-29 02:13:53.000000000 +0200
@@ -198,6 +198,14 @@
     return [ super initWithContentRect:contentRect styleMask:styleMask backing:backingType defer:flag ];
 }
 
+- (BOOL)performKeyEquivalent:(NSEvent*)event
+{
+	/* give the menu a chance to handle the key equivalent */
+	[[NSApp mainMenu] performKeyEquivalent:event];
+	/* avoid beep by pretending we handled it */
+	return YES;	
+}
+
 @end
 
 @implementation SDL_QuartzWindowDelegate
