diff -ru SDL_image-1.2.10.org/Makefile.in SDL_image-1.2.10/Makefile.in
--- SDL_image-1.2.10.org/Makefile.in	2009-11-12 07:42:10.000000000 +0100
+++ SDL_image-1.2.10/Makefile.in	2009-12-18 00:57:29.000000000 +0100
@@ -385,7 +385,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libSDL_image.la: $(libSDL_image_la_OBJECTS) $(libSDL_image_la_DEPENDENCIES) 
-	$(libSDL_image_la_LINK) -rpath $(libdir) $(libSDL_image_la_OBJECTS) $(libSDL_image_la_LIBADD) $(LIBS)
+	$(libSDL_image_la_LINK) -rpath $(libdir) $(libSDL_image_la_OBJECTS) $(libSDL_image_la_LIBADD)
 
 clean-noinstPROGRAMS:
 	@list='$(noinst_PROGRAMS)'; test -n "$$list" || exit 0; \
