diff -ru tcl8.5.11.org/generic/tcl.h tcl8.5.11/generic/tcl.h
--- tcl8.5.11.org/generic/tcl.h	2011-11-01 15:04:51.000000000 +0100
+++ tcl8.5.11/generic/tcl.h	2012-01-29 16:31:41.000000000 +0100
@@ -168,7 +168,7 @@
  *       MSVCRT.
  */
 
-#if (defined(__WIN32__) && (defined(_MSC_VER) || (__BORLANDC__ >= 0x0550) || defined(__LCC__) || defined(__WATCOMC__) || (defined(__GNUC__) && defined(__declspec))))
+#if (defined(__WIN32__) && (defined(_MSC_VER) || (defined(__BORLANDC__) && (__BORLANDC__ >= 0x0550)) || defined(__LCC__) || defined(__WATCOMC__) || (defined(__GNUC__) && defined(__declspec))))
 #   define HAVE_DECLSPEC 1
 #   ifdef STATIC_BUILD
 #       define DLLIMPORT
diff -ru tcl8.5.11.org/generic/tclPosixStr.c tcl8.5.11/generic/tclPosixStr.c
--- tcl8.5.11.org/generic/tclPosixStr.c	2011-11-01 15:04:51.000000000 +0100
+++ tcl8.5.11/generic/tclPosixStr.c	2012-01-29 16:34:52.000000000 +0100
@@ -341,7 +341,7 @@
 #ifdef EPERM
     case EPERM: return "EPERM";
 #endif
-#if defined(EPFNOSUPPORT) && (!defined(ENOLCK) || (ENOLCK != EPFNOSUPPORT))
+#if !defined(_MSC_VER) && defined(EPFNOSUPPORT) && (!defined(ENOLCK) || (ENOLCK != EPFNOSUPPORT))
     case EPFNOSUPPORT: return "EPFNOSUPPORT";
 #endif
 #ifdef EPIPE
@@ -401,7 +401,7 @@
 #ifdef ESHUTDOWN
     case ESHUTDOWN: return "ESHUTDOWN";
 #endif
-#ifdef ESOCKTNOSUPPORT
+#if !defined(_MSC_VER) && defined ESOCKTNOSUPPORT
     case ESOCKTNOSUPPORT: return "ESOCKTNOSUPPORT";
 #endif
 #ifdef ESPIPE
@@ -789,7 +789,7 @@
 #ifdef EPERM
     case EPERM: return "not owner";
 #endif
-#if defined(EPFNOSUPPORT) && (!defined(ENOLCK) || (ENOLCK != EPFNOSUPPORT))
+#if !defined(_MSC_VER) && defined(EPFNOSUPPORT) && (!defined(ENOLCK) || (ENOLCK != EPFNOSUPPORT))
     case EPFNOSUPPORT: return "protocol family not supported";
 #endif
 #ifdef EPIPE
@@ -849,7 +849,7 @@
 #ifdef ESHUTDOWN
     case ESHUTDOWN: return "can't send after socket shutdown";
 #endif
-#ifdef ESOCKTNOSUPPORT
+#if !defined(_MSC_VER) && defined ESOCKTNOSUPPORT
     case ESOCKTNOSUPPORT: return "socket type not supported";
 #endif
 #ifdef ESPIPE
diff -ru tcl8.5.11.org/unix/configure tcl8.5.11/unix/configure
--- tcl8.5.11.org/unix/configure	2011-11-04 13:47:57.000000000 +0100
+++ tcl8.5.11/unix/configure	2012-01-29 16:35:53.000000000 +0100
@@ -14925,7 +14925,7 @@
 else
 
 	    if test "$cross_compiling" = yes; then
-  tcl_cv_strtod_buggy=buggy
+  tcl_cv_strtod_buggy=ok
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
diff -ru tcl8.5.11.org/unix/Makefile.in tcl8.5.11/unix/Makefile.in
--- tcl8.5.11.org/unix/Makefile.in	2011-11-04 13:47:57.000000000 +0100
+++ tcl8.5.11/unix/Makefile.in	2012-01-29 16:36:18.000000000 +0100
@@ -732,7 +732,7 @@
 	fi
 	@EXTRA_INSTALL_BINARIES@
 
-install-libraries: libraries $(INSTALL_TZDATA) install-msgs
+install-libraries: libraries $(INSTALL_TZDATA)
 	@for i in "$(INCLUDE_INSTALL_DIR)" "$(SCRIPT_INSTALL_DIR)"; \
 	    do \
 	    if [ ! -d "$$i" ] ; then \
diff -ru tcl8.5.11.org/win/Makefile.in tcl8.5.11/win/Makefile.in
--- tcl8.5.11.org/win/Makefile.in	2011-11-04 13:47:58.000000000 +0100
+++ tcl8.5.11/win/Makefile.in	2012-01-29 16:36:39.000000000 +0100
@@ -601,7 +601,7 @@
 	    $(COPY) $(REG_LIB_FILE) $(LIB_INSTALL_DIR)/reg1.2; \
 	    fi
 
-install-libraries: libraries install-tzdata install-msgs
+install-libraries: libraries
 	@for i in $(prefix)/lib $(INCLUDE_INSTALL_DIR) \
 		$(SCRIPT_INSTALL_DIR); \
 	    do \
