diff -ru tcl8.5.18.orig/win/Makefile.in tcl8.5.18/win/Makefile.in
--- tcl8.5.18.orig/win/Makefile.in	2015-03-06 16:13:58.000000000 +0100
+++ tcl8.5.18/win/Makefile.in	2015-10-04 22:14:02.109235175 +0200
@@ -606,7 +606,7 @@
 	    $(COPY) $(REG_LIB_FILE) $(LIB_INSTALL_DIR)/reg$(REGDOTVER); \
 	    fi
 
-install-libraries: libraries install-tzdata install-msgs
+install-libraries: libraries
 	@for i in $(prefix)/lib $(INCLUDE_INSTALL_DIR) \
 		$(SCRIPT_INSTALL_DIR); \
 	    do \
