diff -ru tcl8.5.7.org/generic/tcl.h tcl8.5.7/generic/tcl.h
--- tcl8.5.7.org/generic/tcl.h	2009-04-10 18:54:51.000000000 +0200
+++ tcl8.5.7/generic/tcl.h	2009-08-27 17:38:59.000000000 +0200
@@ -170,7 +170,7 @@
  *       MSVCRT.  
  */
 
-#if (defined(__WIN32__) && (defined(_MSC_VER) || (__BORLANDC__ >= 0x0550) || defined(__LCC__) || defined(__WATCOMC__) || (defined(__GNUC__) && defined(__declspec))))
+#if (defined(__WIN32__) && (defined(_MSC_VER) || (defined(__BORLANDC__) && (__BORLANDC__ >= 0x0550)) || defined(__LCC__) || defined(__WATCOMC__) || (defined(__GNUC__) && defined(__declspec))))
 #   define HAVE_DECLSPEC 1
 #   ifdef STATIC_BUILD
 #       define DLLIMPORT
