diff -ru tcl8.5.8.org/generic/tcl.h tcl8.5.8/generic/tcl.h
--- tcl8.5.8.org/generic/tcl.h	2009-11-03 20:21:37.000000000 +0100
+++ tcl8.5.8/generic/tcl.h	2010-01-02 14:16:24.000000000 +0100
@@ -170,7 +170,7 @@
  *       MSVCRT.  
  */
 
-#if (defined(__WIN32__) && (defined(_MSC_VER) || (__BORLANDC__ >= 0x0550) || defined(__LCC__) || defined(__WATCOMC__) || (defined(__GNUC__) && defined(__declspec))))
+#if (defined(__WIN32__) && (defined(_MSC_VER) || (defined(__BORLANDC__) && (__BORLANDC__ >= 0x0550)) || defined(__LCC__) || defined(__WATCOMC__) || (defined(__GNUC__) && defined(__declspec))))
 #   define HAVE_DECLSPEC 1
 #   ifdef STATIC_BUILD
 #       define DLLIMPORT
