diff -ru zlib-1.2.3.org//Makefile.in zlib-1.2.3/Makefile.in
--- zlib-1.2.3.org//Makefile.in	2005-07-18 04:25:21.000000000 +0200
+++ zlib-1.2.3/Makefile.in	2011-01-16 16:08:32.000000000 +0100
@@ -33,7 +33,7 @@
 SHAREDLIBV=libz.so.1.2.3
 SHAREDLIBM=libz.so.1
 
-AR=ar rc
+AR=ar
 RANLIB=ranlib
 TAR=tar
 SHELL=/bin/sh
@@ -68,7 +68,7 @@
 	fi
 
 libz.a: $(OBJS) $(OBJA)
-	$(AR) $@ $(OBJS) $(OBJA)
+	$(AR) rc $@ $(OBJS) $(OBJA)
 	-@ ($(RANLIB) $@ || true) >/dev/null 2>&1
 
 match.o: match.S
