diff -ru zlib-1.2.8.orig/configure zlib-1.2.8/configure
--- zlib-1.2.8.orig/configure	2013-03-24 06:30:09.000000000 +0100
+++ zlib-1.2.8/configure	2013-05-18 18:22:58.000000000 +0200
@@ -192,9 +192,9 @@
         EXE='.exe' ;;
   MINGW* | mingw*)
 # temporary bypass
-        rm -f $test.[co] $test $test$shared_ext
-        echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
-        leave 1
+#        rm -f $test.[co] $test $test$shared_ext
+#        echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
+#        leave 1
         LDSHARED=${LDSHARED-"$cc -shared"}
         LDSHAREDLIBC=""
         EXE='.exe' ;;
